variables.tf

```diff
@@ -62,6 +62,31 @@ variable "additional_security_group_ids" {
   default     = []
 }
 
+variable "detailed_monitoring" {
+  description = "Enable detailed monitoring of instance"
+  type = bool
+  default = true
+}
+
+variable "associate_public_ip_address" {
+  description = "Associate public IP address"
+  type = bool
+  # default should fall back to subnet setting
+  default = null
```

Comment:
I'm not sure this would work, see [the doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#associate_public_ip_address):

> Boolean value, can be left unset.

Could you please elaborate on the default fall back to the subnet setting?

Reply:
So I believe it's Terraform v0.12+ allows setting `null` to represent unset. I used default = null here to keep the default as the unset option and not require a user to set this variable.

My understanding is that if this isn't set, it'll use the property of the subnet on whether to assign a public IP address. We have public and private subnets in our VPC, on our public subnets they have "Associate public IPv4 address" set to true. When I tried this change with `null`, the nodes came up with a public IP address. I haven't tried using it in our private subnets, but my guess is they would not get a public IP there.

Reply:
Thank you for clarifying! That makes sense.
Yeah, I see now that this attribute has type `nullable.TypeNullableBool` in the [provider](https://github.com/hashicorp/terraform-provider-aws/blob/98ef8a66e117b5216aab733bfdd2779e9c7e398c/internal/service/ec2/ec2_launch_template.go#L742C1-L747C9), and allows to set the attribute to null explicitly.